log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java

```diff
@@ -100,59 +242,191 @@ private String targetMessageTest(final Message message) {
+     */
+    @PluginBuilderFactory
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A {@link RegexFilter} builder instance.
+     */
+    public static final class Builder extends AbstractFilterBuilder<RegexFilter.Builder>
+            implements org.apache.logging.log4j.core.util.Builder<RegexFilter> {
+
+        /* NOTE: LOG4J-3086 - No patternFlags in builder - this functionality has been deprecated/removed. */
+
+        /**
+         * The regular expression to match.
+         */
+        @PluginBuilderAttribute
+        @Required(message = "No 'regex' provided for RegexFilter")
+        private @Nullable String regex;
+
+        /**
+         * If {@code true}, for {@link ParameterizedMessage}, {@link StringFormattedMessage},
+         * and {@link MessageFormatMessage}, the message format pattern; for {@link StructuredDataMessage},
+         * the message field will be used as the match target.
+         */
+        @PluginBuilderAttribute
+        private @Nullable Boolean useRawMsg;
+
+        /** Private constructor. */
+        private Builder() {
+            super();
+        }
+
+        /**
+         * Sets the regular-expression.
+         *
+         * @param regex the regular-expression
+         * @return this builder
+         */
+        public Builder setRegex(final String regex) {
+            this.regex = Assert.requireNonEmpty(regex, "The 'regex' attribute must not be null or empty.");
+            return this;
+        }
+
+        /**
+         * Sets the use raw msg flag.
+         *
+         * @param useRawMsg {@code true} if the message format-patter/field will be used as match target;
+         *                  otherwise, {@code false}
+         * @return this builder
+         */
+        public Builder setUseRawMsg(final boolean useRawMsg) {
+            this.useRawMsg = useRawMsg;
+            return this;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public boolean isValid() {
+            return (Strings.isNotEmpty(this.regex));
+        }
+
+        /**
+         * Builds and returns a {@link RegexFilter} instance configured by this builder.
+         *
+         * @return the created {@link RegexFilter} or {@code null} if the builder is misconfigured
+         */
+        @Override
+        public @Nullable RegexFilter build() {
+
+            // validate the "regex" attribute
+            if (Strings.isEmpty(this.regex)) {
+                LOGGER.error("Unable to create RegexFilter: The 'regex' attribute be set to a non-empty String.");
+                return null;
+            }
+
+            // build with *safety* to not throw exceptions
+            try {
+                return new RegexFilter(this);
+            } catch (final Exception ex) {
+                LOGGER.error("Unable to create RegexFilter. {}", ex.getMessage(), ex);
+                return null;
+            }
         }
```

Comment:
We had [a `dev@` discussion about this](https://lists.apache.org/thread/h2oydyk6xld47ljttqvflbt4530o73vw), and the conclusion is to throw exception on failures while building components. Would you mind updating this `build()` logic accordingly, please?